Numerical_model/krusty.py

```diff
@@ -0,0 +1,270 @@
+        dt = 0.01                           # Time step size
+        tsec = np.arange(0, Ttime+dt, dt)   # time vector
+        rho = [rho0]                        # value vectors
+        Tf = [Tf0]
+        n = []
+        c = []
+        n = [self.n0]
+        c = self.c
+        order = 4                           # order of approach
+        for ind, t in enumerate(tsec[1:]):
+            # initial values
+            n_j0 = n[-1]
+            c_j0 = [x[-1] for x in c]
+            T_j0 = Tf[-1]
+            rho_j0 = rho[-1]
+            # print(t,n_j0,T_j0,rho_j0)
+            rho_j = self.frho(rho0,T_j0,n_j0)
+            if rho_j <= float(0):
+                rho_j = 0
+            # first derivative
+            dna = self.fn(n_j0, c_j0, rho_j0)
+            dca = [f for f in self.fc(n_j0, c_j0, rho_j0)]
+            dTa = self.fTemp(n_j0, T_j0)
+            #
+            n_j1 = n_j0 + dna*dt/2
+            c_j1 = [c + dc*dt/2 for c, dc, in zip(c_j0, dca)]
+            T_j1 = T_j0 + dTa*dt/2
+            #
+            dnb = self.fn(n_j1, c_j1, rho_j)
+            dcb = [f for f in self.fc(n_j1, c_j1, rho_j)]
+            dTb = self.fTemp(n_j1, T_j1)
+            #
+            n_j2 = n_j0 + dnb*dt/2
+            c_j2 = [c + dc*dt/2 for c, dc, in zip(c_j0, dcb)]
+            T_j2 = T_j0 + dTb*dt/2
+            #
+            dnc = self.fn(n_j2, c_j2, rho_j)
+            dcc = [f for f in self.fc(n_j2, c_j2, rho_j)]
+            dTc = self.fTemp(n_j2, T_j2)
+            #
+            n_j3 = n_j0 + dnc*dt
+            c_j3 = [c + dc*dt for c, dc, in zip(c_j0, dcc)]
+            T_j3 = T_j0 + dTc*dt/2
+            #
+            dnd = self.fn(n_j3, c_j3, rho_j)
+            dcd = [f for f in self.fc(n_j3, c_j3, rho_j)]
+            dTd = self.fTemp(n_j3, T_j3)
+            #
+            n_j4 = n_j0 + (dna + 2*dnb + 2*dnc + dnd)*dt/6
+            c_j4 = [cj + (a + 2*b + 2*c + d)*dt/6 for cj, a, b,
+                    c, d in zip(c_j0, dca, dcb, dcc, dcd)]
+            T_j4 = T_j0 + (dTa + 2*dTb + 2*dTc + dTd)*dt/6
+            # print(t,n_j4,rho[-1])
+            n.append(n_j4)
+            Tf.append(T_j4)
+            for i in range(self.groups):
+                c[i].append(c_j4[i])    
+            rho.append(rho_j)
+
+        tmin = np.divide(tsec,60)
+        plt.figure(1)
+        plt.plot(tmin,n)
+        plt.title('Neutron Density')
+        plt.xlabel('Time [sec]')
+        plt.ylabel('n(t)')
+        plt.yscale('log')
+        plt.figure(2)
+        plt.plot(tsec,rho)
+        plt.title('Reactivity')
+        plt.xlabel('Time [sec]')
+        plt.ylabel(r'$\rho(t)$')
+        plt.yscale('log')
+        plt.figure(3)
+        plt.plot(tsec,Tf)
+        plt.title('Fuel Temperature')
+        plt.xlabel('Time [sec]')
+        plt.ylabel(r'T_f(t)')
+        plt.yscale('linear')
+        # plt.yscale('log')
+        plt.show()
+
+    def Heun(self):
+        rho_cost = float(0.6)              # Reactivity [$]
```

Comment:
You are repeating this step in runga kutta and heun, can you save these as part of the overall class? Just saves some redundancy